### PR TITLE
Verify document generation

### PR DESCRIPTION
**Pull request by chaitra-mylarappachar**
Re-arranged, Re-named files in examples/ so that the document generation tool (tfplugindocs) picks default examples during the generation of docs.
The examples/ folder is structured based on the recommendation here, 
- https://github.com/hpe-hcss/hpegl-terraform-resources-template#add-example-config-to-examplesresources-and-examplesdata-sources
- https://github.com/hashicorp/terraform-plugin-docs#conventional-paths.

### CODE REVIEW

**Comment by neilgierman** [comment]
Do we need some `.github/workflows` created or will that be a follow up PR?

**Comment by chaitra-mylarappachar** [comment]
> Do we need some `.github/workflows` created or will that be a follow up PR?

Yes, will take care of that next.

File: internal/resources/resource_volume.go [hunk 1]
```diff
@@ -125,7 +125,8 @@ func VolumeResource() *schema.Resource {
 			State: schema.ImportStatePassthrough,
 		},
 
-		Schema: volumeSchema(),
+		Schema:      volumeSchema(),
+		Description: "Provides Volume resource. This allows creation, deletion and updation of Metal volumes.",
```
Comment (mchuang3) [inline internal/resources/resource_volume.go:129]:
```suggestion
		Description: "Provides Volume resource. This allows creation, deletion and update of Metal volumes.",
```

File: internal/resources/resource_ssh_key.go [hunk 2]
```diff
@@ -41,8 +41,8 @@ func SshKeyResource() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
-
-		Schema: sshKeySchema(),
+		Schema:      sshKeySchema(),
+		Description: "Provides SSH resource. This allows creation, deletion and updation of Metal SSHKeys",
```
Comment (mchuang3) [inline internal/resources/resource_ssh_key.go:45]:
```suggestion
		Description: "Provides SSH resource. This allows creation, deletion and update of Metal SSHKeys",
```

File: internal/resources/resource_host.go [hunk 3]
```diff
@@ -225,7 +225,8 @@ func HostResource() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
-		Schema: hostSchema(),
+		Schema:      hostSchema(),
+		Description: "Provides Host resource. This allows Metal Host creation, deletion and updation.",
```
Comment (mchuang3) [inline internal/resources/resource_host.go:229]:
```suggestion
		Description: "Provides Host resource. This allows Metal Host creation, deletion and update.",
```

File: internal/resources/resource_host.go [hunk 4]
```diff
@@ -10,8 +10,8 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
-	rest "github.com/hpe-hcss/quake-client/v1/pkg/client"
 	"github.com/HewlettPackard/hpegl-metal-terraform-resources/pkg/client"
+	rest "github.com/hpe-hcss/quake-client/v1/pkg/client"
```
Comment (mchuang3) [inline internal/resources/resource_host.go:14]: I thought this was public repo.  Should we eliminate hpe-hcss references?

File: internal/resources/resource_network.go [hunk 5]
```diff
@@ -190,7 +190,8 @@ func ProjectNetworkResource() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
-		Schema: networkSchema(),
+		Schema:      networkSchema(),
+		Description: "Provides Network resource. This allows creation, deletion and updation of Metal networks.",
```
Comment (mchuang3) [inline internal/resources/resource_network.go:194]:
```suggestion
		Description: "Provides Network resource. This allows creation, deletion and update of Metal networks.",
```

File: internal/resources/resource_project.go [hunk 6]
```diff
@@ -149,7 +149,8 @@ func ProjectResource() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
-		Schema: projectSchema(),
+		Schema:      projectSchema(),
+		Description: "Provides Project resource. This allows creation, deletion and updation of Metal projects.",
```
Comment (mchuang3) [inline internal/resources/resource_project.go:153]:
```suggestion
		Description: "Provides Project resource. This allows creation, deletion and update of Metal projects.",
```

File: examples/data-sources/hpegl_metal_usage/data-source.tf [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  default = "2020-01-13T07:44:00Z"
+}
+
+variable "end" {
+  default = "2020-04-13T07:44:00Z"
+}
+
+data "hpegl_metal_usage" "used" {
+  start = var.start
+  #end = var.end
```
Comment (mchuang3) [inline examples/data-sources/hpegl_metal_usage/data-source.tf:13]: Is end optional?  Add comment or remove?

File: internal/resources/datasource_available_resources.go [hunk 8]
```diff
@@ -199,7 +199,8 @@ func existingVolumeResource() *schema.Resource {
 
 func DataSourceAvailableResources() *schema.Resource {
 	return &schema.Resource{
-		Read: dataSourceAvailableResourcesRead,
+		Read:        dataSourceAvailableResourcesRead,
+		Description: "Provides a lists of available resources in a project for creating Hosts and Volumes.",
```
Comment (mchuang3) [inline internal/resources/datasource_available_resources.go:203]:
```suggestion
		Description: "Provides a list of available resources in a project for creating Hosts and Volumes.",
```

File: examples/data-sources/hpegl_metal_usage/data-source.tf (hunk 7 above)
Comment (chaitra-mylarappachar) [reply examples/data-sources/hpegl_metal_usage/data-source.tf]: Yes, optional. Removed it.

File: internal/resources/resource_host.go (hunk 4 above)
Comment (chaitra-mylarappachar) [reply internal/resources/resource_host.go:14]: Still haven't moved the rest-client portion to this new Org. Will remove this after that. Just keeping it, for now, to be able to build and test the ongoing changes.

**Review by mchuang3 — APPROVED** [review]

**Pull request merged**